extensions/kafka/src/main/java/io/deephaven/kafka/ingest/GenericRecordVectorFieldCopier.java

```diff
@@ -0,0 +1,132 @@
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_BYTE : (byte) o;
+                        ++i;
+                    }
+                    return new ByteVectorDirect(out);
+                };
+            }
+            // There is no "SHORT" in Avro.
+
+            if (componentType.equals(int.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return IntVectorDirect.ZERO_LEN_VECTOR;
+                    }
+                    final int[] out = new int[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_INT : (int) o;
+                        ++i;
+                    }
+                    return new IntVectorDirect(out);
+                };
+            }
+            if (componentType.equals(long.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return LongVectorDirect.ZERO_LEN_VECTOR;
+                    }
+                    final long[] out = new long[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_LONG : (long) o;
+                        ++i;
+                    }
+                    return new LongVectorDirect(out);
+                };
+            }
+            if (componentType.equals(float.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return FloatVectorDirect.ZERO_LEN_VECTOR;
+                    }
+                    final float[] out = new float[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_FLOAT : (float) o;
+                        ++i;
+                    }
+                    return new FloatVectorDirect(out);
+                };
+            }
+            if (componentType.equals(double.class)) {
+                return (GenericArray<?> ga) -> {
+                    final int gaSize = ga.size();
+                    if (gaSize == 0) {
+                        return DoubleVectorDirect.ZERO_LEN_VECTOR;
+                    }
+                    final double[] out = new double[gaSize];
+                    int i = 0;
+                    for (Object o : ga) {
+                        out[i] = (o == null) ? NULL_DOUBLE : (double) o;
+                        ++i;
+                    }
+                    return new DoubleVectorDirect(out);
+                };
+            }
+            if (componentType.equals(boolean.class) || componentType.equals(String.class)) {
```

Comment:
Booleans can't be `null`? Or the `toArray` will address it?

Reply:
New code for that that gets the type right also.